File: week1/mario/mario.c
#include <cs50.h>
#include <stdio.h>

int main (void)
{
    int height;
    do
    {
        height = get_int("Height:\n");
    }
    while(height>8 || height < 1);

    int width = height * 2 + 2;
    int h;
    int w;

    //outer loop
    for(h = 0; h<height; h++)
    {

        // inner loop
        for(w = 0; w<width; w++)
        {


            if(w < height -h - 1 || (height - 1  < w && w < height + 2))
            {
                printf(" ");
            }
           else if(w> width - height + h)
            {
                break;
            }
           else
           {
               printf("#");
           }


        }
        printf("\n");
    }

}

File: week6/sentimental-mario-more/mario.py
# main function


def main():
    while True:
        try:
            height = int(input("Height:"))
            if height < 9 and height > 0:
                block(height)
                return
        except ValueError:
            print("Please input a number")

# function that arranges blocks


def block(height):
    width = height * 2 + 2
    # outer loop
    for h in range(0, height):

        # inner loop
        for w in range(0, width):

            if w < (height - h - 1) or ((height - 1 < w) and (w < height + 2)):
                print(" ", end='')
            elif w > (width - height + h):
                print("")
                break
            else:
                print("#", end='')

        #print(" ")
    print("")


main()